pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -944,6 +978,28 @@ func (c *Controller) WorkloadInstanceHandler(si *model.WorkloadInstance, event m
 	}
 }
 
+func (c *Controller) onNamespaceEvent(obj interface{}, ev model.Event) error {
+	var nw string
+	if ev != model.EventDelete {
+		ns, ok := obj.(*v1.Namespace)
+		if !ok {
+			log.Warnf("Namespace watch getting wrong type in event: %T", obj)
+			return nil
+		}
+		nw = ns.Labels[label.IstioNetwork]
+	}
+	c.Lock()
+	oldDefaultNetwork := c.network
+	c.network = nw
+	c.Unlock()
```

Comment:
Can remove the lock

Reply:
Can we though? I've seen races around setting networkForRegistry in UTs. This is no different.

Reply:
This field is string type, in theory we donot need to protect it. The w and r sequence does not matter much

controller.RWMutex is meant to protect the maps

Reply:
I am not sure that is correct based on https://stackoverflow.com/questions/51249918/immutability-of-string-and-concurrency, we need to protect it

Reply:
From the last answer, when to write `network` concurrently, we should protect it, otherwise golang race detector fails.